experimental/dds/tree2/src/feature-libraries/editable-tree-2/editableTreeTypes.ts

```diff
@@ -297,8 +297,25 @@ export interface MapNode<in out TSchema extends MapSchema> extends TreeNode {
 		thisArg?: any,
 	): void;
 
-	// TODO: Add `set` method when FieldKind provides a setter (and derive the type from it).
-	// set(key: string, content: FlexibleFieldContent<TSchema["mapFields"]>): void;
+	/**
+	 * Sets a new value associated with `key`.
+	 * @param key - The key under which the new value will be set.
```

Comment:
If you like, you can more or less copy paste the doc for this method from JS Map, though rolling your own is fine too if you think it's better.

Reply:
That's a good call. Will update.

Reply:
I updated to use slightly modified versions of the JS Map API docs.